### PR TITLE
refactor: speller from C++ fp to D delegate

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dsymbol.d [hunk 1]
```diff
@@ -614,9 +614,29 @@ public:
...
+             * Doing this first is a lot faster.
+             */
+            size_t len = strlen(seed);
+            if (!len)
+                return null;
+            Identifier id = Identifier.lookup(seed, len);
+            if (!id)
+                return null;
+            cost = 0;
+            Dsymbol s = this;
```
Comment (yebblies) [inline src/dsymbol.d]: Don't need `s` any more.

**Comment by yebblies** [comment]
- Speller has a header file which will need updating/deleting.
- Is there a good reason to avoid using inline function literals?  They let us avoid writing out the arg types.  eg `speller( (arg, arg) { ... })`
- Most of `root.speller` should be made `private`.

File: src/dsymbol.d (hunk 1 above)
Comment (WalterBright) [inline src/dsymbol.d]: I did that to minimize the refactoring changes and make it easier to review :-)

**Comment by WalterBright** [comment]
I structured it this way to minimize the changes, to make it easier to review. Certainly more could be done. But recall our last discussion on doing too much in one PR :-)

**Comment by WalterBright** [comment]
Removed speller.h

**Comment by WalterBright** [comment]
BTW, I find non-trivial function literals to be rather hard to follow. Making them a distinct function is more natural.

**Comment by 9rnsr** [comment]
Auto-merge toggled on
